flash/1.2.11/Dockerfile

```diff
@@ -0,0 +1,50 @@
+ARG FLASH_VER="1.2.11"
+
+FROM ubuntu:focal as builder
+ARG FLASH_VER
+
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    wget \
+    zlib1g-dev
+
+RUN wget http://ccb.jhu.edu/software/FLASH/FLASH-$FLASH_VER.tar.gz && \
+ tar -xzf FLASH-$FLASH_VER.tar.gz && \
+ rm FLASH-$FLASH_VER.tar.gz
+
+WORKDIR /FLASH-$FLASH_VER
+RUN make
+
+FROM ubuntu:focal as app
+ARG FLASH_VER
+
+# Labels need to go in the production stage
+LABEL base.image="ubuntu:focal"
+LABEL dockerfile.version="1"
+LABEL software="FLASH"
+LABEL software.version=$FLASH_VER
+LABEL description="FLASH (Fast Length Adjustment of Short Reads): merges overlapping paired-end reads"
+LABEL website="http://ccb.jhu.edu/software/FLASH"
+LABEL license="https://github.com/ebiggers/flash/blob/master/COPYING"
+
+# Copy only executable into production stage to reduce image size
+COPY --from=builder FLASH-$FLASH_VER/flash /usr/local/bin/flash
+
+WORKDIR /data
+
+FROM app as test
+ARG FLASH_VER
+
+# Get test dependencies
+RUN apt-get update && apt-get install -y \
+    bc \
+    python3 \
+    wget
```

Comment:
why are you installing wget twice?

Reply:
I install it twice because of the three-stage Dockerfile - the goal is to make the `app` stage as slim as possible. `builder` needs wget to get and compile the program, `app` doesn't need wget (only the program executable), and `test` does need wget again to fetch test data. This ends up saving space in the `app` stage:
<img width="666" alt="Screen Shot 2022-04-04 at 9 35 02 AM" src="https://user-images.githubusercontent.com/30396464/161495638-116e6a44-7f4f-4a3e-932c-8184bbe5602d.png">

Reply:
Sizes are from building the image to each stage, i.e. `docker build --target=builder -t flash-builder flash/1.2.11/`